expenser-server/middleware/auth.go

```
@@ -16,37 +14,33 @@ type errorResponse struct {
+				return
+			}
+
+			userID, err := auth.ValidateJWT(token, []byte(jwtSecret))
+			if err != nil {
+				logger.Error("error with token", map[string]interface{}{"error": err})
+				respondWithJson(w, err)
+				return
+			}
+
+			// Add userID to request context
+			ctx := context.WithValue(r.Context(), "userID", userID)
```

golangci-lint (1.62.2)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)